### ISSUE TITLE
`InstanceVariables` doesn't work with if statements

### ISSUE THREAD

**Issue by ghiculescu** [sds/haml-lint#341]
This test, if added [here](https://github.com/sds/haml-lint/blob/master/spec/haml_lint/linter/instance_variables_spec.rb), fails

```ruby
      context 'in an if statement' do
        let(:haml) { <<-HAML }
          - if @greeting
            Hello
        HAML

        it { should report_lint line: 1 }
      end
```

With this output:

```
  1) HamlLint::Linter::InstanceVariables when the file name matches the matcher and there is an instance variable in an if statement is expected to report a lint on line 1
     Failure/Error: it { should report_lint line: 1 }
       expected that a lint would be reported on line 1
     # ./spec/haml_lint/linter/instance_variables_spec.rb:59:in `block (5 levels) in <top (required)>'
```

**Comment by ghiculescu** [comment]
This is because `HamlLint::Tree::SilentScriptNode#parsed_script` is returning `nil`, I guess because `"if @greeting"` is an incomplete Ruby statement 🤔

**Comment by sds** [comment]
Fixed in #364.
